### PR TITLE
reference operator on deployed objects

### PR DESCRIPTION
**Pull request by phoracek**
With this patch, we extend all deployed objects with an annotation
referencing version of the operator that deployed them. This should
make debugging easier. For instance, it should allow us to see whether
there are any leftovers from older operator versions.

### CODE REVIEW

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go [hunk 1]
```diff
@@ -287,6 +293,16 @@ func (r *ReconcileNetworkAddonsConfig) renderObjects(networkAddonsConfig *opv1al
 	}
 	objs = append([]*unstructured.Unstructured{applied}, objs...)
 
+	// Label objects with version of the operator they were created by
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		labels["networkaddonsoperator.network.kubevirt.io/version"] = operatorVersion
```
Comment (qinqon) [inline pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: You can take the prefix of the string from here https://github.com/kubevirt/cluster-network-addons-operator/blob/bdc8d36b05e55c084fb19fd0ed96e8359b960ab3/pkg/apis/networkaddonsoperator/v1alpha1/register.go#L13

Comment (phoracek) [reply pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: Done

**Comment by booxter** [comment]
Can we test it too?

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go [hunk 2]
```diff
@@ -287,6 +293,16 @@ func (r *ReconcileNetworkAddonsConfig) renderObjects(networkAddonsConfig *opv1al
 	}
 	objs = append([]*unstructured.Unstructured{applied}, objs...)
 
+	// Label objects with version of the operator they were created by
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		labels[opv1alpha1.SchemeGroupVersion.Group] = operatorVersion
```
Comment (qinqon) [inline pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: opv1alpha1.SchemeGroupVersion.Group + "/version"

**Comment by phoracek** [comment]
@qinqon fixed, please take a look

**Pull request merged**